src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -43,47 +38,28 @@
 
     static final Logger LOGGER = Logging.getLogger(ComplexGeoJsonWriter.class);
 
-    private static Class<?> NON_FEATURE_TYPE_PROXY;
     private static final String DATATYPE = "@dataType";
     /**
      * A string constant for representing a not needed key name because object is being added inside
      * an already named json array
      */
     private static final String INSIDE_ARRAY_ATTRIBUTE = "${inside-array}";
 
-    static {
-        try {
-            NON_FEATURE_TYPE_PROXY =
-                    Class.forName("org.geotools.data.complex.config.NonFeatureTypeProxy");
-        } catch (ClassNotFoundException e) {
-            // might be ok if the app-schema datastore is not around
-            if (StreamSupport.stream(
-                            Spliterators.spliteratorUnknownSize(
-                                    DataStoreFinder.getAllDataStores(), Spliterator.ORDERED),
-                            false)
-                    .anyMatch(
-                            f ->
-                                    f != null
-                                            && f.getClass()
-                                                    .getSimpleName()
-                                                    .equals("AppSchemaDataAccessFactory"))) {
-                LOGGER.log(
-                        Level.FINE,
-                        "Could not find NonFeatureTypeProxy yet App-schema is around, probably the class changed name, package or does not exist anymore",
-                        e);
-            }
-            NON_FEATURE_TYPE_PROXY = null;
-        }
-    }
-
     private final GeoJSONBuilder jsonWriter;
 
     private boolean geometryFound = false;
     private CoordinateReferenceSystem crs;
     private long featuresCount = 0;
 
-    public ComplexGeoJsonWriter(GeoJSONBuilder jsonWriter) {
+    private ComplexGeoJsonWriterOptions settings;
```

Comment:
Looks like this is a hard dependency, please make it a final field.

Reply:
fixed